libcst/metadata/scope_provider.py

```diff
@@ -647,6 +647,7 @@ def __init__(self, provider: "ScopeProvider") -> None:
             Union[cst.Call, cst.Annotation, cst.Subscript]
         ] = set()
         self.__in_ignored_subscript: Set[cst.Subscript] = set()
+        self.__ignore_annotation: bool = False
```

Comment:
This should probably be a stack or a counter instead of a simple bool to keep track of nesting. Let's add a nested test case while we're at it.